packages/cfd/src/Containers/cfd-password-change.tsx

```diff
@@ -0,0 +1,195 @@
+
+type TCFDPasswordFormReusedProps = {
+    platform: typeof CFD_PLATFORMS[keyof typeof CFD_PLATFORMS];
+    error_message: string;
+    validatePassword: (values: TCFDPasswordFormValues) => FormikErrors<TCFDPasswordFormValues>;
+};
+
+type TCFDPasswordChangeProps = TCFDPasswordFormReusedProps & {
+    error_type?: string;
+    form_error?: string;
+    onCancel: () => void;
+    onForgotPassword: () => void;
+    setNewPasswordValue?: React.Dispatch<React.SetStateAction<string>>;
+    should_set_trading_password: boolean;
+};
+
+type TOnSubmitPasswordChange = (
+    values: TCFDPasswordFormChangeValues,
+    actions: FormikHelpers<TCFDPasswordFormChangeValues>
+) => Promise<void>;
+
+const CFDPasswordChange = observer(
+    ({
+        error_type,
+        form_error,
+        onCancel,
+        onForgotPassword,
+        setNewPasswordValue,
+        should_set_trading_password,
+    }: TCFDPasswordChangeProps) => {
+        const { ui, modules } = useStore();
+        const { cfd } = modules;
+        const { setIsMt5PasswordChangedModalVisible } = cfd;
+        const { is_mobile } = ui;
+        const has_cancel_button = (isDesktop() ? !should_set_trading_password : true) || error_type === 'PasswordReset';
+
+        const handleCancel = () => {
+            if (!has_cancel_button) {
+                return undefined;
+            }
+            if (should_set_trading_password) {
+                return onCancel();
+            }
+
+            return onForgotPassword();
+        };
+
+        const validate = (values: FormikValues) => {
+            const errors: FormikErrors<{
+                old_password: string;
+                new_password: string;
+            }> = {};
+
+            if (!values.old_password) {
+                errors.old_password = localize('This field is required');
+            }
+            if (!values.new_password) {
+                errors.new_password = localize('This field is required');
+            }
+
+            if (validatePassword(values.new_password)) errors.new_password = validatePassword(values.new_password);
+
+            return errors;
+        };
+
+        const handleSubmit: TOnSubmitPasswordChange = async (values, actions) => {
+            const response = await WS.tradingPlatformPasswordChange({
+                old_password: values.old_password,
+                new_password: values.new_password,
+                platform: CFD_PLATFORMS.MT5,
+            });
+
+            if (response.error) {
+                if (response.error.code === 'PasswordError')
+                    actions.setFieldError('old_password', response.error.message);
+                if (response.error.code === 'InputValidationFailed')
+                    actions.setFieldError(
+                        'new_password',
+                        'Please include at least 1 special character such as ( _ @ ? ! / # ) in your password.'
```

Comment:
I think we should localize the validation error message as well before setting it

Reply:
thought of this but missed as it is custom error